packages/firebase_data_connect/firebase_data_connect/lib/src/firebase_data_connect.dart

```diff
@@ -54,24 +54,19 @@ class FirebaseDataConnect extends FirebasePluginPlatform {
   void checkTransport() {
     transportOptions ??=
         TransportOptions('firebasedataconnect.googleapis.com', null, true);
-    transport = getTransport(transportOptions!, options, auth, appCheck);
+    transport = getTransport(
+        transportOptions!, options, app.options.appId, auth, appCheck);
   }
 
   /// Returns a [QueryRef] object.
   QueryRef<Data, Variables> query<Data, Variables>(
       String operationName,
       Deserializer<Data> dataDeserializer,
-      Serializer<Variables>? varsSerializer,
+      Serializer<Variables> varsSerializer,
```

Comment:
Considering the varSerializer` has been made a required arg here as well, we should probably aim to merge this one first: https://github.com/firebase/flutterfire/pull/13386 so it is in the CHANGELOG.